core/runtime/src/main/java/io/quarkus/runtime/CommandLineRuntimeConfig.java

```diff
@@ -0,0 +1,23 @@
+package io.quarkus.runtime;
+
+import java.util.Optional;
+
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigPhase;
+import io.quarkus.runtime.annotations.ConfigRoot;
+
+/**
+ * This configuration class is here to avoid warnings when using {@code -Dquarkus.args=...}.
```

Comment:
is this really the right thing to do ?  

maybe we should rename it to something different at maven level so there is no collision ? 

having args as just a string seems counter to how we otherwise treat args in rest of quarkus (as a String[] or eventually a List<String>). ?

Reply:
We do that a lot.

As for string vs list, see the comment, I can't be more explicit.

As for all the other knobs we introduced to avoid the warnings, they are not supposed to be used. That's what we decided a long time ago rather than having an ignore list (which we had at some point).

We could introduce a special converter to split the things in the same way Maven is doing things but I don't think it's worth it. And we would probably have to copy Maven code to be exactly on par.

BTW, I think we need that PR for 1.4 so blocking the PR just for that seems counterproductive.